File: LeetCode Practice Problems/804- Unique Morse Code Words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        map<string,int> m;
        string v[] = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        for(string s : words)
        {
            string res = "";
            for(char c : s)
            {
                res = res + v[c-'a'];
            }
            m[res]++;
            cout<<res<<" ";
        }
        
        return m.size();
    }
};

File: LeetCode/LeetCode Practice Problems/804-Unique Morse Code Words.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String a[] = new String[]{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        
        HashSet<String> s = new HashSet();
        
        for(String word: words) {
            String cur = "";
            for(char ch: word.toCharArray()) {
                cur = cur + a[ch - 'a'];
            }
            s.add(cur);
        }
        
        return s.size();
    }
}
